iocore/eventsystem/SocketManager.cc

```diff
@@ -108,3 +111,36 @@ SocketManager::close(int s)
   } while (res == -EINTR);
   return res;
 }
+
+bool
+SocketManager::fastopen_supported()
```

Comment:
Fast open support is a property of the running kernel, not of the Traffic Server build. As @bryancall points out, it also needs to be enabled by a sysctl, so there's no reliable compile time test for this. I figured that doing the test at reconfigure time strikes a balance between getting it right and the performance cost of continually checking.